.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,46 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
```

Comment:
does not make sense to run on `main` only, if we actually use `development/2.1` as well...

Reply:
added in fixup commit for the specific changes: https://github.com/scality/osis/pull/143/commits/7050fcb1546cee061594fdc81b7fc9bbbbe7d161

Reply:
Updated link: https://github.com/scality/osis/pull/143/commits/2c118c55d4631a63a2ed87d9986b25a18c0f4c93